### PR TITLE
Querying: Support runtime registered data source

### PR DESCRIPTION
**Pull request by torkelo**
This is another attempt at https://github.com/grafana/scenes/pull/45 but instead solving it by allowing scene library consumers to register an instance of DataSourceApi. 

Trying to make it possible to issue custom API call queries that result in DataFrames but via the SceneQueryRunner 

Advantages to this over a customQueryHandler passed to SceneQueryRunner: 

* [x] You can now use mixed data source queries to merge data from Prometheus and to your custom data source  (without adding customQueryHandler to the DataQuery model or creating a new CompositeQueryRunner) 
* [x] Docs 
* [x] Demo scene

### CODE REVIEW

File: packages/scenes-app/src/demos/flexLayout.tsx [hunk 1]
```diff
@@ -25,10 +26,11 @@ export function getFlexLayoutTest(defaults: SceneAppPageState) {
               children: [
                 new SceneFlexItem({
                   minWidth: '70%',
-                  body: PanelBuilders.timeseries()
-                    .setTitle('Dynamic height and width')
-                    .setData(getQueryRunnerWithRandomWalkQuery({}, { maxDataPointsFromWidth: true }))
-                    .build(),
+                  body: new VizPanel({
```
Comment (bfmatei) [inline packages/scenes-app/src/demos/flexLayout.tsx:29]: Out of curiosity, what's the reason for changing from builder to declarative here?

**Review by bfmatei — COMMENTED** [review]
This is pure awesomeness, ty @torkelo. I'll leave some comments here and there.

File: packages/scenes-app/src/demos/flexLayout.tsx (hunk 1 above)
Comment (torkelo) [reply packages/scenes-app/src/demos/flexLayout.tsx:29]: did not really need it here, was not setting any visualization settings

**Review by dprokop — APPROVED** [review]
Looks good - apart from that one test failing! 

Adding @imatwawana for docs review.

**Review by bfmatei — APPROVED** [review]
Looks fine from my pov as well.

**Pull request merged**